### DISCUSSION TITLE
Hook the data fetch request.

### DISCUSSION THREAD

**Discussion by GordonSmith** [duckdb/duckdb-wasm#1776] [Q&A] [answered]
Is there any way to hook an URL request so I can manually fetch and format the data?  I want my user to be able to enter something like this:

```sql
    CREATE TABLE direct AS
        SELECT * FROM 'xxx, yyy'
```

Then I would like to intercept the fetch in my function:

```ts
function fetchJson(url) {
  //  Call some API passing in xxx, yyy
  //  Filter and format the data for loading
  //  Return the data back into the AsyncDuckDB instance
}
```

IOW I don't want to make the API call until the user specifically requests it.

**Comment by aszenz** [comment, marked as answer]
Yes u can use the function getTableNames to get the table names required to run a query and then load them before running the actual query. 

https://shell.duckdb.org/docs/classes/index.AsyncDuckDBConnection.html#getTableNames

**Reply by GordonSmith** [reply]
That works well - thx!!!
